src/main/java/com/pi4j/example/components/LEDMatrix.java

```diff
@@ -0,0 +1,202 @@
+     * @param Columns   How many columns of LED
+     * @param brightness How bright the leds can be at max, Range 0 - 1
+     */
+    public LEDMatrix(Context pi4j, int Rows, int Columns, double brightness) {
+        this(pi4j, new int[Rows][Columns], brightness, DEFAULT_CHANNEL);
+    }
+
+    /**
+     * Creates a new simpleLed component with a custom BCM pin.
+     *
+     * @param pi4j    Pi4J context
+     * @param matrix How many LEDs are on this Strand
+     * @param brightness How bright the leds can be at max, Range 0 - 255
+     * @param channel which channel to use
+     */
+    public LEDMatrix(Context pi4j, int[][] matrix, double brightness, int channel) {
+        this.matrix = matrix;
+        this.brightness = brightness;
+        this.context = pi4j;
+
+        // Allocate SPI transmit buffer (same size as PCM)
+        this.numLeds = 0;
+        for (int[] ints : matrix) {
+            this.numLeds += ints.length;
+        }
+
+        this.ledStrip = new LEDStrip(pi4j, numLeds, brightness, channel);
+        this.spi = ledStrip.spi;
+    }
+
+    /**
+     * @return the pi4j context
+     */
+    public Context getContext() {
+        return this.context;
+    }
+
+    /**
+     * Setting all LEDS off and closing the strip
+     */
+    public void close() {
+        ledStrip.close();
+    }
+
+    /**
+     * function to get the amount of the leds in the matrix
+     *
+     * @return int with the amount of leds over all
+     */
+    public int getNumPixels() {
+        return numLeds;
+    }
+
+    /**
+     * function to get the amount of the leds in the specified strip
+     *
+     * @return int with the amount of leds over all
+     */
+    public int getNumPixels(int strip) {
+        if (strip > matrix.length || strip < 0) {
+            throw new IllegalArgumentException("the strip specified does not exist");
+        }
+        return matrix[strip].length;
+    }
+
+    /**
+     * function to get the color (as an int) of a specified led
+     *
+     * @param pixel which position on the ledstrip, range 0 - numLEDS-1
+     * @return the color of the specified led on the strip
+     */
+    public int getPixelColor(int strip, int pixel) {
+        if (strip > matrix.length || strip < 0 || pixel > matrix[strip].length || pixel < 0) {
+            throw new IllegalArgumentException("the strip or led specified does not exist");
+        }
+        return matrix[strip][pixel];
+    }
+
+    /**
+     * setting the color of a specified led on the strip
+     *
+     * @param pixel which position on the strip, range 0 - numLEDS-1
+     * @param color the color that is set
+     */
+    public void setPixelColor(int strip, int pixel, int color) {
+        if (strip > matrix.length || strip < 0 || pixel > matrix[strip].length || pixel < 0) {
+            throw new IllegalArgumentException("the strip or led specified does not exist");
+        }
+        matrix[strip][pixel] = color;
+    }
+
```

Comment:
new function set/get PixelColor(int pixelNumber, int color);

to address a led by its number in the led strip

Reply:
created